src/Microsoft.Kusto.ServiceLayer/DataSource/DataSourceBase.cs

```diff
@@ -108,27 +89,16 @@ public async Task<T> ExecuteScalarQueryAsync<T>(string query, CancellationToken
 
         public abstract string GenerateExecuteFunctionScript(string functionName);
 
+        public abstract void UpdateAzureToken(string azureToken);
+
         /// <inheritdoc/>
         public DataSourceType DataSourceType { get; protected set; }
-
```

Comment:
ClusterName, Database, and SchemaState are obfuscated to KustoClient.cs. They have readonly properties and mutator methods.